### PR TITLE
#40. Fix signUsingHmacSha1()

### PR DESCRIPTION
**Pull request by Aerendir**
Add an ` if`statement to avoid the superflous adding of an ampersand (&) if `secret_token` isn't provided/used.

See Issue #40.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "guzzlehttp/oauth-subscriber",
+    "name": "aerendir/oauth-subscriber",
```
Comment (mtdowling) [inline composer.json]: ?

Comment (Aerendir) [inline composer.json]: Ooops... Fixed, I think...

Comment (GrahamCampbell) [inline composer.json]: you can download your fork without publishing it to packagist, infact it's better to do that

Comment (GrahamCampbell) [inline composer.json]: kind of abusing packagist otherwise

Comment (Aerendir) [inline composer.json]: I used the GitHub repo

File: src/Oauth1.php [hunk 2]
```diff
@@ -239,8 +239,12 @@ private function prepareParameters($data)
      */
     private function signUsingHmacSha1($baseString)
     {
-        $key = rawurlencode($this->config['consumer_secret'])
-            . '&' . rawurlencode($this->config['token_secret']);
+        $key = rawurlencode($this->config['consumer_secret']);
+
+        // Add token only if present to avoid wrong encoding due to the superflous ampersand (&)
+        if (isset($this->config['token_secret']) && !empty($this->config['token_secret'])) {
```
Comment (kkopachev) [inline src/Oauth1.php]: just `!empty` is enough

Comment (Aerendir) [inline src/Oauth1.php]: @kkopachev,  you are right, `empty` is enough.

Comment (Aerendir) [inline src/Oauth1.php]: I've closed this "dirty" pull request and submitted a new one, using only `empty` and without the "exceding" commits. Hope this is good...

**Pull request closed without merging**